### PR TITLE
ht: fix gcc7 compilation

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/ht/default.nix [hunk 1]
```diff
@@ -15,6 +15,8 @@ stdenv.mkDerivation rec {
 
   hardeningDisable = [ "format" ];
 
+  patches = [ ./gcc7.patch ];
```
Comment (grahamc) [inline pkgs/applications/editors/ht/default.nix]: Forgot the patch :)

**Comment by Mic92** [comment]
Here we go.

**Comment by Mic92** [comment]
@GrahamcOfBorg build ht

**Comment by Mic92** [comment]
also backported a32fcfd0dad

**Pull request merged**